src/snowcli/zipper.py

```diff
@@ -39,15 +39,17 @@ def add_file_to_existing_zip(zip_file: str, file: str):
         myzip.write(file, os.path.basename(file))
 
 
-def zip_dir(source: Path, dest_zip: Path) -> None:
+def zip_dir(
+    source: Path, dest_zip: Path, mode: Literal["r", "w", "x", "a"] = "w"
+) -> None:
     files_to_pack: Iterator[Path] = filter(
         _to_be_zipped, map(lambda f: f.absolute(), source.glob("**/*"))
     )
 
-    with ZipFile(dest_zip, "w", ZIP_DEFLATED, allowZip64=True) as package_zip:
+    with ZipFile(dest_zip, mode, ZIP_DEFLATED, allowZip64=True) as package_zip:
         for file in files_to_pack:
             log.debug("Adding %s to %s", file, dest_zip)
-            package_zip.write(file, arcname=file.relative_to(source.absolute()))
+            package_zip.write(file, arcname=os.path.relpath(file, source))
```

Comment:
What was the issue with `file.relative_to`?

Reply:
Long story - it was changed at least three times in the meantime. Got back to original.